gadget/mountedfilesystem.go

```diff
@@ -696,53 +709,53 @@ func (f *mountedFilesystemUpdater) backupOrCheckpointFile(dstRoot, source, targe
 
 	// TODO: enable support for symlinks when needed
 	if osutil.IsSymlink(dstPath) {
-		return false, fmt.Errorf("cannot backup file %s: symbolic links are not supported", target)
+		return false, "", fmt.Errorf("cannot backup file %s: symbolic links are not supported", target)
 	}
 
 	if !osutil.FileExists(dstPath) {
 		// destination does not exist and will be created when writing
 		// the udpate, no need for backup
-		return true, nil
+		return true, "", nil
 	}
 
 	if osutil.FileExists(backupName) {
 		// file already checked and backed up
-		return true, nil
+		return true, backupName, nil
 	}
 	if osutil.FileExists(sameStamp) {
 		// file already checked, same as the update, move on
-		return false, nil
+		return false, dstPath, nil
```

Comment:
it's a bit unclear to me given that it will go unused whether it makes sense to return something for originalDataPath if willWrite is false,  maybe this should be:

false, "", nil ?

explained in the comment I mentioned before.

Same for all the false cases below.

Reply:
different take: this also mean that we could just return the backup path, and when is "" it means nothing is getting written?

Reply:
Not exactly, there is still a case when there the destination did not exist prior to the update, thus there's no backup path, but the file still gets written.

Reply:
ah, the  `true, "", nil` cases?  my first point is still valid though?

Reply:
Yes, I've tweaked the code and it returns "" when the file will not be written (added a doc comment too).